caput/mpiarray.py

```diff
@@ -735,6 +735,9 @@ def redistribute(self, axis: int) -> "MPIArray":
         if self.axis == axis or self.comm is None:
             return self
 
+        if self.comm.size == 1:
```

Comment:
If you want something pedantic, here you go:
```suggestion
        # Avoid repeat mpi property calls
        csize = self.comm.size
        crank = self.comm.rank
        
        if csize == 1:
```
(Moving lines 759-762 up here to save yourself yet another property call.)